docs/source/trainer_cli.rst

```diff
@@ -0,0 +1,275 @@
+    python trainer.py --config config.yaml
+
+The call to the :class:`~pytorch_lightning.utilities.trainer_cli.TrainerCli`
+class takes care of parsing command line and config file options, instantiating
+the classes, setting up a callback to save the config in the log directory and
+finally running :func:`trainer.fit`.
+
+After multiple trainings with different configurations, a previous run can be
+trivially reproduced by using the config in the respective log directory, e.g.:
+
+.. code-block:: bash
+
+    python trainer.py --cfg lightning_logs/version_7/config.yaml
+
+If a separate :class:`~pytorch_lightning.core.datamodule.LightningDataModule`
+class is required, the trainer tool just needs a small modification as follows:
+
+.. testcode::
+
+    from pytorch_lightning.utilities.trainer_cli import TrainerCli
+
+    TrainerCli(MyModel, MyDataModule)
+
+The start of a possible implementation of :class:`MyModel` including the
+recommended argument descriptions in the docstring could be the one below. Note
+that by using type hints and docstrings there is no need to duplicate this
+information to define its configurable arguments.
+
+.. code-block:: python
+
+    class MyModel(LightningModule):
+
+        def __init__(
+            self,
+            encoder_layers: int = 12,
+            decoder_layers: List[int] = [2, 4]
+        ):
+            """Example encoder-decoder model
+
+            Args:
+                encoder_layers: Number of layers for the encoder
+                decoder_layers: Number of layers for each decoder block
+            """
+            ...
+
+With this model class, the help of the trainer tool would look as follows:
+
+.. code-block:: bash
+
+    $ python trainer.py --help
+    usage: trainer.py [-h] [--print_config] [--config CONFIG]
+                      [--trainer.logger LOGGER]
+                      ...
+
+    pytorch-lightning trainer command line tool
+
+    optional arguments:
+      -h, --help            show this help message and exit
+      --print_config        print configuration and exit
+      --config CONFIG       Path to a configuration file in json or yaml format.
+                            (default: null)
+
+    Customize every aspect of training via flags:
+      ...
+      --trainer.max_epochs MAX_EPOCHS
+                            Stop training once this number of epochs is reached.
+                            (type: int, default: 1000)
+      --trainer.min_epochs MIN_EPOCHS
+                            Force training for at least these many epochs (type: int,
+                            default: 1)
+      ...
+
+    Example encoder-decoder model:
+      --model.encoder_layers ENCODER_LAYERS
+                            Number of layers for the encoder (type: int, default: 12)
+      --model.decoder_layers DECODER_LAYERS
+                            Number of layers for each decoder block (type: List[int],
+                            default: [2, 4])
```

Comment:
not sure if this keeps here as it means we have to keep it updated and as it not linked to the API with test it would be easy to forget...

Reply:
I proposed adding a snippet of the help. I think it helps users to understand what jsonargparse provides.

I don't think the attributes {max,min}_epochs will change much, it is worth having it to me personally.

Reply:
Is there any way we could generate this content dynamically?

Reply:
we only include it as a quick and tiny view of what's provided. I don't think we need to go through that effort